app/controllers/loans_controller.rb

```diff
@@ -1,25 +1,95 @@
 class LoansController < ApplicationController
   before_action :set_loan, only: %i[ show edit update destroy ]
 
-  # GET /loans or /loans.json
-  def index
-    @loans = Loan.all
+   def index
+    
+      @loans = current_user.loans.all
+      
+      # Initialize Variable
+      @pmt_schedule_chart = [];   
+      @adj_pmt_schedule_chart = []; 
+      @data = []; 
+      @data_adj = []; 
+      @total_interest = 0;
+      @total_adj_interest = 0;
+      @pay_off_period = [];
+      @pay_off_period_adj = [];
```

Comment:
It's best practice that [controllers should be "skinny" and models should be "fat"](https://riptutorial.com/ruby-on-rails/example/9609/fat-model--skinny-controller#example). This means, controllers should only be in charge of preparing the instance variables required for the view template. All other logic that is required to generate those variables should be done in the models whenever possible.

With that in mind, all of the loops you have in this controller should be extracted into part of some method.

One approach to extracting the code required for these variables is to make a method that calculates each one.

So that your controller ends up looking something like:

```rb
# app/controllers/loans_controller.rb
...
@pmt_schedule_chart = @loans.calculate_pmt_schedule_chart
```

```rb
# app/models/loan.rb

class Loan < ApplicationRecord
  # ...
  def self.calculate_pmt_schedule_chart
    self.order(:created_at).each do |loan|
      # ...
    end
  end
end
```

where `self` in this case will be whatever `@loans` is. [This might be a nice reading to complement](https://stackoverflow.com/a/21681483).

I imagine you could have a class method that calculates each of these instance variables that you need.

How does that sound?

Reply:
Looking at this again, are `@data` and `@data_adj` being used in the view template? If not, you should be able to remove them.

Reply:
I think `@data` and `@data_adj` are legacy from a different graphing gem I did not end up using requiring a different format, so those can be deleted to clean up the controller/model.  I think since this is a larger change, I will finish up some of the simpler updates, then open up a new pull request and new review for moving these calculations around.